### PR TITLE
[MM-21373] Don't send verification

### PR DESCRIPTION
**Pull request by hahmadia**
#### Summary

If require email verification is off in the system console, when a user signs up, we shouldn't be sending a verification link in the email.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-21373

#### Screenshot (Disregard broken image links, using inbucket which doesn't render it correctly)
image

### CODE REVIEW

**Comment by srkgupta** [comment]
@hahmadia 
Please note that only the Welcome Email Notification should not be sent when Email Notification and Email Verification is disabled. If Email Verification is enabled, irrespective of the setting Email Notification, the email should always be sent.

**Comment by thefactremains** [comment]
@srkgupta We might be saying the same thing, but I want to be sure. There are 2 issues. `Enable Email Notifications` and `Require Email Verification` should not be linked (MM-21682) and this one - where the verification was being sent - even though `Require Email Verification` was disabled. 

The direction provided to @hahmadia was to remove any link or dependency on these settings. 

You commented:
> An additional check needs to be added to check if the email notifications are enabled and only then emails should be sent.

Can you please help me understand the reasoning behind that?

**Comment by srkgupta** [comment]
@thefactremains 
Currently when the email notifications are disabled, system should not be sending any email notification to users. However currently the user gets an email notification (as shown in the PR description). Ideally this email should not be sent

File: app/email.go [hunk 1]
```diff
@@ -170,6 +170,10 @@ func (a *App) SendSignInChangeEmail(email, method, locale, siteURL string) *mode
 }
 
 func (a *App) SendWelcomeEmail(userId string, email string, verified bool, locale, siteURL string) *model.AppError {
+	if !*a.Config().EmailSettings.SendEmailNotifications {
```
Comment (jespino) [inline app/email.go]: I'm not sure, but I think this email should be sent if this is for verification but not for welcome. If you require email verification, but doesn't have email notifications, is normal to not receive the welcome message, but the email verification is needed for the application functionality, not for notifying the user. Any opinion on this @wiersgallak?

Comment (hahmadia) [reply app/email.go]: @jespino Already discussed this with @thefactremains . He mentioned that it should still be sent even if Email Notifications is off. I made a mistake. Here is the possibilities chart
```
Email Verification Setting | Email Notification Setting
0 0 - No Email Sent
1 0 - Email with Verification link sent
0 1 - Welcome Email (no verification link) sent
1 1 - Email with Verification link sent
```
I will fix this shortly and push commit.

**Comment by hahmadia** [comment]
@srkgupta I have updated it. I have spoken with @thefactremains and these are the possibilities and what should happen for each one
```
Email Verification Setting | Email Notification Setting
0 0 - No Email Sent
1 0 - Email with Verification link sent
0 1 - Welcome Email (no verification link) sent
1 1 - Email with Verification link sent
```

**Pull request merged**